doc/entropy-seems-wrong.md

```diff
@@ -0,0 +1,107 @@
+
+
+Calculation of Entropy
+----------------------
+
+Researchers can spend their entire lives devising schemes to measure the amount of entropy in a given data set.  In essence, one only measures how much entropy there *isn't* instead of how much there *is*.
+
+
+### Entropy Basics
+
+For our examples, we will use 8 character long passwords and primarily restrict ourselves to the 26 lowercase letters in the English language.
+
+Assuming that you roll dice or have another truly random method for picking letters, you will generate one of a possible 208,827,064,576 passwords.  That number is generated because you have 26 different letters you can pick for each of the 8 letters in the password.  The math to compute the number of permutations would be 26^8, or 208 billion.
+
+This includes passwords like "thimbles" and "stupider", but more on that weakness later.
+
+To compute the number of bits of entropy, one just tries to figure out how many possibilities there are (208 billion) and then figures out how big of a binary number it would take to represent it.  To do that you use `log(26^8) / log(2)` becase normally the `log()` function provides results in base 10 and we require them in base 2.  This is also known as `log2(26^8)`, and the result is about 37.6 bits of entropy.
+
+
+### Character Set Sizes
+
+This math is possible because we know how many letters are in our alphabet.  To include uppercase letters and lowercase letters the set grows to `26 + 26 = 52` options for each letter in the password.  If we were dealing with an unknown language or if we didn't want to group the letters into sets of characters then we have a harder time.
+
+There are a couple options that the password strength software has taken to combat this problem.  The first is using a technique from Claude Shannon, where the letters in the password itself are used as the character set size.  Using the password of "aaaabbbb" would indicate there's only 2 options per character and we'd have `2^8` possibilities.  This works way better for longer passwords that have more varied characters.
+
+The other technique is to lump letters together into groups.  When a letter is encountered from a group, the entire group's size is included.  A lot of systems require you to use "one capital letter, one number, one symbol" and our character sets are similar.  So the password of "aA0!aA0!" would score poorly with Shannon's technique (`4^8` possibilities) but much better when using the groups of letters (`26 + 26 + 10 + 10 = 72` options for each character, `72^8` possibilities).
+
+
+### Character Prediction
+
+When checking passwords, one should be encouraged (though not required) to use randomly generated passwords that do not look like words.  If you typically use words, those passwords should be weaker.  To accomplish this, a large amount of text was analyzed for letter trigraphs, which is essentially saying "when given two letters, what's the chance you can predict the third letter?"
```

Comment:
If the quote is part of a sentence you need "..." at the beginning to denote there was more before the quoted part.